tutorials/foreign-object.html

```diff
@@ -0,0 +1,813 @@
+    attrs: {
+        body: {
+            width: 'calc(w)',
+            height: 'calc(h)',
+            fill: {
+                type: 'linearGradient',
+                stops: [
+                    { offset: 0, color: '#ff5c69' },
+                    { offset: 0.5, color: '#ff4252' },
+                    { offset: 1, color: '#ed2637' }
+                ]
+            }
+        },
+        foreignObject: {
+            width: 'calc(w-12)',
+            height: 'calc(h-12)',
+            x: 6,
+            y: 6
+        }
+    },
+}, {
+    // The /* xml */ comment is optional.
+    // It is used to tell the IDE that the markup is XML.
+    markup: joint.util.svg/* xml */`
+        &lt;style&gt;
+            div.card__background {
+                background-color: #131e29;
+                height: 100%;
+            }
+            p.card__text {
+                color: #F2F2F2;
+                font: 16px sans-serif;
+                padding: 10px;
+                margin: 0;
+            }             
+        &lt;/style&gt;
+        &lt;rect @selector="body"/&gt;
+        &lt;foreignObject @selector="foreignObject"&gt;
+            &lt;div @selector="background"
+                xmlns="http://www.w3.org/1999/xhtml"
+                class="card__background"
+            &gt;
+                &lt;p @selector="text"
+                    class="card__text"
+                &gt;
+                    Lorem ipsum dolor sit amet, consectetur adipiscing elit.
+                &lt;/p&gt;
+            &lt;/div&gt;
+        &lt;/foreignObject&gt;
+    `
+});
+</code></pre>
+
+        <h3>A more interactive example</h3>
+
+        <p>
+            One typical method which diagram libraries utilize to allow users work with HTML is to render HTML on top of an underlying element.
+            While this approach can be successful, it comes with a lot of additional complexity such as the need to keep dimensions and position 
+            of the HTML in sync with the element itself which isn't ideal. It's possible to use this approach with JointJS, and still to this day,
+            this is how many diagram libraries provide this functionality, especially if they use
+            HTML <a href="https://developer.mozilla.org/en-US/docs/Web/HTML/Element/canvas" target="_blank">canvas</a>.
+        </p>
+
+        <p>
+            As JointJS primarily works with SVG, and SVG was designed to work well with other web standards such as HTML, taking advantage of
+            <code>foreignObject</code> can allow users of JointJS create HTML-rich elements while avoiding some of the difficulties of other 
+            approaches.
+        </p>
+
+        <p>
+            Creating a static card-like element was a gentle introduction to working with <code>foreignObject</code>, but if we are being realistic,
+            a lot of users will most likely want to work with more interactive elements. A <code>form</code> element is a section which usually 
```

Comment:
I would remove "but if we are being realistic, a lot of users will most likely want to work with more interactive elements".

Maybe just say: "...introduction to working with `foreignObject`. For a more realistic example, let's try to create a `form` element with several interactive controls."